pkg/cmd/render/certs.go

```diff
@@ -0,0 +1,69 @@
+package render
+
+import (
+	"crypto/x509/pkix"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/openshift/library-go/pkg/crypto"
+	"k8s.io/apiserver/pkg/authentication/user"
+)
+
+// keyMaterial simplifies handling of the key data produced by createKeyMaterial
+type keyMaterial struct {
+	caCert     []byte
+	caKey      []byte
+	caBundle   []byte
+	clientCert []byte
+	clientKey  []byte
+}
+
+// createKeyMaterial returns the key material for a self-signed CA, its CA bundle,
+// and a client cert signed by the CA.
+func createKeyMaterial(signerName, clientName string) (*keyMaterial, error) {
+	// CA
+	caSubject := pkix.Name{CommonName: signerName, OrganizationalUnit: []string{"openshift"}}
+	caExpiryDays := 10 * 365 // 10 years
+	signerCAConfig, err := crypto.MakeSelfSignedCAConfigForSubject(caSubject, caExpiryDays)
+	if err != nil {
+		return nil, err
+	}
+	caCert, caKey, err := signerCAConfig.GetPEMBytes()
+	if err != nil {
+		return nil, err
+	}
+
+	// Bundle
+	caBundle, err := crypto.EncodeCertificates(signerCAConfig.Certs[0])
+	if err != nil {
+		return nil, err
+	}
+
+	// Client cert
+	ca := &crypto.CA{
+		Config:          signerCAConfig,
+		SerialGenerator: &crypto.RandomSerialGenerator{},
+	}
+	clientUser := &user.DefaultInfo{
+		Name:   clientName,
+		UID:    uuid.New().String(),
+		Groups: []string{clientName},
+	}
+	clientCertDuration := time.Hour * 24 * 365 * 10 // 10 years
```

Comment:
should 10 years be a constant or at least use the same definition?

https://github.com/openshift/cluster-etcd-operator/pull/546/files#diff-28893b89756831caa4aa8af970a87b34173bdfe9a6edb86daaf948b7a6bc03d3R26

Reply:
re: reuse I'm assuming you mean 10 years in days? I'm usually pretty DRY but I figured inlining a single-use definition would be easier to read.

Note that this long duration is unlikely to persist given our intent to support rotation.